modules/internal/String.chpl

```diff
@@ -482,11 +482,11 @@ module String {
     inline proc size return len;
 
     /*
-      :returns: The number of Unicode codepoints in the string.
+      :returns: The number of codepoints in the string (typically Unicode).
       */
-    proc ulength {
+    proc codePointLength {
```

Comment:
For strings, length makes much more sense to me than size, because most people use size to mean the number of bytes allocated to the string object, and length to mean the number of characters in the string - two different things.

However, we've unfortunately made string.size a synonym for string.length, so I'll create a string.codePointSize that is a synonym for string.codePointLength.

Reply:
>  so I'll create a string.codePointSize that is a synonym for string.codePointLength.

I'm not sure it's worth going that far. If you (and other reviewers) feel that `codePointLength` is the right choice, I'm comfortable with it - just wanted to make sure `codePointSize` was considered.

> we've unfortunately made string.size a synonym for string.length, 

Not sure how I feel about it today, but here's the discussion around this decision: https://github.com/chapel-lang/chapel/issues/8395

Reply:
I'm happy with Length vs Size but `codePointLength` sounds like the length of a codepoint (e.g. 4 bytes always).

lengthInCodePoints would be more reasonable, if verbose. What about `numCodePoints` or `numberOfCodePoints` ?

Reply:
I like `numCodePoints` out of the options mentioned so far.

Reply:
So do I.  I will make that change.